AssertionError in FloatCanvas ScaledBitmap zooming

**Operating system**: Windows 11
**wxPython version & source**: 4.2.3 (pypi)
**Python version & source**: 3.12

**Description of the problem**: A wxAssertionError (C++ assertion failure) occurs when zooming in or out on a ScaledBitmap in FloatCanvas beyond certain limits -- either too far in or out.

For zooming up:
```
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 452, in WheelEvent
    self.GUIMode.OnWheel(event)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\GUIMode.py", line 132, in OnWheel
    self.Canvas.Zoom(1.1, point, centerCoords = "Pixel", keepPointInPlace=True)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 758, in Zoom
    self.MoveImage(-delta, 'World')
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 718, in MoveImage
    self.Draw()
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 625, in Draw
    self._DrawObjects(dc, self._DrawList, ScreenDC, self.ViewPortBB, HTdc)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 985, in _DrawObjects
    Object._Draw(dc, WorldToPixel, ScaleWorldToPixel, HTdc)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FCObjects.py", line 2157, in _Draw
    Img = self.Image.Scale(int(W), int(H))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
wx._core.wxAssertionError: C++ assertion ""data"" failed at ..\..\src\common\image.cpp(535) in wxImage::ResampleNearest(): unable to create image
```
For zooming down:
```
Traceback (most recent call last):
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 452, in WheelEvent
    self.GUIMode.OnWheel(event)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\GUIMode.py", line 130, in OnWheel
    self.Canvas.Zoom(0.9, point, centerCoords = "Pixel", keepPointInPlace=True)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 758, in Zoom
    self.MoveImage(-delta, 'World')
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 718, in MoveImage
    self.Draw()
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 625, in Draw
    self._DrawObjects(dc, self._DrawList, ScreenDC, self.ViewPortBB, HTdc)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FloatCanvas.py", line 985, in _DrawObjects
    Object._Draw(dc, WorldToPixel, ScaleWorldToPixel, HTdc)
  File "C:\Python312\Lib\site-packages\wx\lib\floatcanvas\FCObjects.py", line 2157, in _Draw
    Img = self.Image.Scale(int(W), int(H))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
wx._core.wxAssertionError: C++ assertion ""(width > 0) && (height > 0)"" failed at ..\..\src\common\image.cpp(450) in wxImage::Scale(): invalid new image size
```

To confrime this issue:
1. Open the wxPython demo and launch the FloatCanvas demo window.
1. From the menu, select Tests > Bitmap Test.
1. In the toolbar, choose either Pan or Zoom.
1. Scroll the mouse wheel up or down many times.

--- a/wx/lib/floatcanvas/FCObjects.py
+++ b/wx/lib/floatcanvas/FCObjects.py
@@ -2139,6 +2139,7 @@ class ScaledBitmap(TextObjectMixin, DrawObject):
         self.CalcBoundingBox()
         self.ScaledBitmap = None
         self.ScaledHeight = None
+        self.MaxSize = 20e6  # cf. FHD 1,920 x 1,080
 
     def CalcBoundingBox(self):
         """Calculate the bounding box."""
@@ -2152,13 +2153,16 @@ class ScaledBitmap(TextObjectMixin, DrawObject):
         XY = WorldToPixel(self.XY)
         H = ScaleWorldToPixel(self.Height)[0]
         W = int(H * (self.bmpWidth / self.bmpHeight))
-        if (self.ScaledBitmap is None) or (H != self.ScaledHeight) :
-            self.ScaledHeight = H
-            Img = self.Image.Scale(W, H)
-            self.ScaledBitmap = wx.Bitmap(Img)
-
         XY = self.ShiftFun(XY[0], XY[1], W, H)
-        dc.DrawBitmap(self.ScaledBitmap, XY, True)
+        if 0 < H * float(W) < self.MaxSize:
+            if (self.ScaledBitmap is None) or (H != self.ScaledHeight):
+                Img = self.Image.Scale(W, H)
+                self.ScaledHeight = H
+                self.ScaledBitmap = wx.Bitmap(Img)
+            dc.DrawBitmap(self.ScaledBitmap, XY, True)
+        else:
+            gc = wx.GraphicsContext.Create(dc)
+            gc.DrawBitmap(self.ScaledBitmap, *XY, W, H)
         if HTdc and self.HitAble:
             HTdc.SetPen(self.HitPen)
             HTdc.SetBrush(self.HitBrush)

